FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,46 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard let self = self else { return }
+
+			guard let (data, response) = try? result.get() else {
+				completion(.failure(RemoteFeedLoader.Error.connectivity))
+				return
+			}
+
+			completion(self.map(response, withData: data))
+		}
+	}
+
+	// MARK: - Helper Methods
+
+	private func map(_ response: HTTPURLResponse, withData data: Data) -> FeedLoader.Result {
+		let decoder = JSONDecoder()
+
+		if response.statusCode == 200, let images = try? decoder.decode(Items.self, from: data) {
+			return .success(images.feedImages)
+		} else {
+			return .failure(RemoteFeedLoader.Error.invalidData)
+		}
+	}
+
+	private struct Items: Codable {
+		private var items: [FeedItem]
```

Comment:
`items` is never mutated so it can be a let:

```suggestion
		private let items: [FeedItem]
```